### DISCUSSION TITLE
Trouble setting up properly to do a build.

### DISCUSSION THREAD

**Discussion by jamesor** [empathicqubit/vscode-cc65-debugger#145] [Q&A] [answered]
I'm trying to set up the debugger and when I try to build the sample program in the template project I get the following issue.

```
Error: Couldn't finish the build successfully. Check the console for details.
  at S.launchRequest
(c:\Users\james\.vscode\extensions\entan-gl.cc65-vice-6.8.1\dist\162.js:1:37458)
```

This is the console output...

```Shell
C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe TARGETS=c64 all
make[1]: Entering directory 'C:/src/vs65-test'
cl65 -t c64 -c --create-dep obj/c64/main.d -Wc "--debug-tables" -Wc "obj/c64/main.tab" -g -o obj/c64/main.o src/main.c
process_begin: CreateProcess(C:\Users\james\.vscode\extensions\entan-gl.cc65-vice-6.8.1\dist\cc65\bin\cl65, cl65 -t c64 -c --create-dep obj/c64/main.d -Wc --debug-tables -Wc obj/c64/main.tab -g 
-o obj/c64/main.o src/main.c, ...) failed.
make (e=193): Error 193
make[1]: *** [Makefile:294: obj/c64/main.o] Error 193
make[1]: Leaving directory 'C:/src/vs65-test'
make: *** [Makefile:326: all] Error 2
```

I've installed the following:
- choco install --version 3.6.1 winvice-nightly
- choco install make
- choco install llvm

I've added the following two lines to the extensions settings.json:
```
    "cc65vice.viceDirectory": "C:/ProgramData/chocolatey/lib/winvice-nightly/tools/GTK3VICE-3.6.1-win64/bin",
    "cc65vice.cc65Home": "C:/Users/james/.vscode/extensions/entan-gl.cc65-vice-6.8.1/dist/cc65"
```

I've also installed the clangd vscode extension.

Can someone help figure out what I might be missing?

**Comment by jamesor** [comment]
I got it working, but not with the VICE installed through Chocolatey.

I downloaded and installed both VICE and CC65 separately. I added both to my user's ENV PATH then removed the 2 path entries I put in the extension's settings.json file.

I couldn't get the VICE/CC65 from the Chocolatey instructions to work. Even trying to run the cl65 command at a power shell prompt didn't work properly. I'm suspecting there was something wrong with the included CC65, or maybe it didn't shortcut the correct binaries to the bin folder for my OS.

**Reply by empathicqubit** [reply]
Are you using a test build of CC65? I don't think the release build supports the features needed for debugging, but it's been a little while since I've tended to this project, so it could have changed.

**Reply by empathicqubit** [reply]
Can you have a look at this template and answer as many questions as you can? You don't need to open a new bug yet, since I'm not sure if it is one https://github.com/empathicqubit/vscode-cc65-debugger/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

For me everything seems to install and run correctly on a clean system with Windows 10, x64.

**Reply by empathicqubit** [reply]
The included cc65 binaries won't go on your path by default. They're meant to be called from the makefile only, but you can definitely put them on your path if you want. This will break whenever a new version is released, however, since the path includes the version number.

**Reply by jamesor** [reply]
I'm running cl65 V2.19 - Git 652949f18 in WSL. The Makefile is building in Linux, but my host is Windows. I have VICE 3.6.1 installed on the host and in the WSL if I type "x64sc.exe" it will launch the emulator. I made a shell script named x64sc which called `x64sc.exe "$@"` and that seamed to launch correctly from the plugin. However, it hung trying to connect to a port, I don't think it was properly passing the prg to launch or had proper port forwarding.

I'm able to get everything working correctly with Windows 10, x64 as long as I don't use a Linux WSL. I only run into trouble when I try to get the Linux subsystem to launch the VICE emulator on the Windows host. This might not be officially supported but thought there might be a way since I saw you were using dev containers in your template.

On Linux Subsystem:
cc65 V2.19 - Git 652949f18 

On Windows Host:
VICE x64sc 3.6.1 (GTK3 3.24.31, GLib 2.70.2, Cairo 1.17.4, Pango 1.50.3)
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion
    ReleaseId    REG_SZ    2009

When I use the launch debug button in VS Code I get this:

```
src/match.s:121: Error: Unresolved external '___bzero'
ld65: Error: 1 unresolved external(s) found - cannot create output file
make[1]: *** [Makefile:317: bin/program.c64] Error 1
make[1]: Leaving directory '/home/user/github/reponame'
make: *** [Makefile:334: all] Error 2
```

However, if I run the Makefile at a command prompt, it builds successfully.

I'll try the version of cc65 you have linked in your bug template and see if that has a difference.

**Reply by empathicqubit** [reply, marked as answer]
WSL isn't supported at the moment. The devcontainer is a docker image containing everything, including the emulator. Maybe if you try running vice inside WSL it will work better? I'm not too sure how well WSL handles GUI apps right now.

**Reply by jamesor** [reply]
When I do a pure Windows 10 setup I have everything working well with the debugger, cc65 and VICE. For now, I'll just stop using WSL. It wasn't really adding much value for this project anyway.
